hack/testing/rsyslog/elasticsearch.conf.j2

```diff
@@ -11,33 +11,44 @@ template(name="index_pattern" type="list") {
     property(name="$!@timestamp" dateFormat="rfc3339" position.from="9" position.to="10")
 }
 
-template(name="id-template" type="string" string="%$!viaq_msg_id%")
-
-if strlen($!viaq_msg_id) == 0 then {
-	# NOTE: depends on rsyslog being compiled with --enable-uuid
-	set $!viaq_msg_id = $uuid;
-}
+template(name="index_template" type="string" string="%$.viaq_index_name%")
+template(name="id_template" type="string" string="%$.viaq_msg_id%")
 
 ruleset(name="error_es") {
-	action(type="omfile" template="viaq_template" file="/var/lib/rsyslog/es-bulk-errors.log")
+    action(type="omfile" template="viaq_template" file="/var/lib/rsyslog/es-bulk-errors.log")
 }
 
 ruleset(name="try_es") {
-	if strlen($.omes!status) > 0 then {
-		# retry case
-		if ($.omes!status == 200) or ($.omes!status == 201) or (($.omes!status == 409) and ($.omes!writeoperation == "create")) then {
-			stop # successful
-		}
-		if ($.omes!writeoperation == "unknown") or (strlen($.omes!error!type) == 0) or (strlen($.omes!error!reason) == 0) then {
-			call error_es
-			stop
-		}
-		if ($.omes!status == 400) or ($.omes!status < 200) then {
-			call error_es
-			stop
-		}
-		# else fall through to retry operation
-	}
+    if strlen($.omes!status) > 0 then {
+        # retry case
+        if ($.omes!status == 200) or ($.omes!status == 201) or (($.omes!status == 409) and ($.omes!writeoperation == "create")) then {
+            stop # successful
+        }
+        if ($.omes!writeoperation == "unknown") or (strlen($.omes!error!type) == 0) or (strlen($.omes!error!reason) == 0) then {
+            call error_es
+            stop
+        }
+        if ($.omes!status == 400) or ($.omes!status < 200) then {
+            call error_es
+            stop
+        }
+        # else fall through to retry operation
+    }
+    if strlen($!viaq_msg_id) > 0 then {
+	    set $.viaq_msg_id = $!viaq_msg_id;
+    } else if strlen($.omes!_id) > 0 then {
+        # retry
+        set $.viaq_msg_id = $.omes!_id;
+    } else {
+	    # NOTE: depends on rsyslog being compiled with --enable-uuid
```

Comment:
Where --enable-uuid is set?  I cannot find it in the spec files...
An indentation issue in line 43? ;)

Reply:
> Where --enable-uuid is set? I cannot find it in the spec files...

It is set by default:
```
>./configure --help|grep uuid
  --enable-uuid           Enable support for uuid generation [default=yes]
```

fixed the indentation issue

Reply:
see also https://kojipkgs.fedoraproject.org//packages/rsyslog/8.36.0/1.fc28/data/logs/x86_64/build.log
```
rsyslog will be compiled with the following settings:
...
    uuid support enabled:                     yes
```

Reply:
I see.  If it's set by default, it's even better.
/lgtm